src/profile-logic/marker-data.js

```diff
@@ -310,6 +310,9 @@ export function correlateIPCMarkers(threads: Thread[]): IPCMarkerCorrelations {
           return 1;
         case 'transferEnd':
           return 2;
+        case 'endpoint':
+        // We don't have a 'phase' field in the older profiles, in that case
+        // their phase is 'endpoint'. (fallthrough)
         default:
           // 'endpoint'
           return 0;
```

Comment:
I believe we can move `return 0` to the `case endpoint` part, and use `throw ensureExhaustiveCheck` here, because `data.phase` is properly typed as an enum.

Reply:
We can't use `ensureExhaustiveCheck` because of two things:
1. `data.phase` is a proper enum but this `default` is also capturing the `undefined` value, becuse for the older profiles, it could be undefined since we don't have this property at all.
2. in the switch-case below, we don't consume all the `phase` enum options. Because we can't have a `transferStart` value for the receiver thread IPC. That's why it's 5 markers and not 6 (3 from sender + 2 from receiver)

Reply:
Mmm right it could be missing. Then I think I'd add an `if` first, or `case undefined`.
But I'll let you decide :-)